app/src/main/java/org/mozilla/focus/locale/screen/LanguageMiddleware.kt

```diff
@@ -0,0 +1,65 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.locale.screen
+
+import android.app.Activity
+import mozilla.components.lib.state.Middleware
+import mozilla.components.lib.state.MiddlewareContext
+import mozilla.components.support.locale.LocaleManager
+import mozilla.components.support.locale.LocaleUseCases
+import org.mozilla.focus.locale.Locales
+import org.mozilla.focus.settings.InstalledSearchEnginesSettingsFragment
+import org.mozilla.gecko.util.ThreadUtils.runOnUiThread
+import java.util.Locale
+
+class LanguageMiddleware(val activity: Activity, private val localeUseCase: LocaleUseCases) :
+    Middleware<LanguageScreenState, LanguageScreenAction> {
+
+    override fun invoke(
+        context: MiddlewareContext<LanguageScreenState, LanguageScreenAction>,
+        next: (LanguageScreenAction) -> Unit,
+        action: LanguageScreenAction
+    ) {
+        when (action) {
+            is LanguageScreenAction.Select -> {
+                LanguageStorage(context = activity).saveCurrentLanguageInSharePref(action.selectedLanguage.tag)
+                setCurrentLanguage(action.selectedLanguage.tag)
```

Comment:
We are not calling `next()` for select? Then this will never end up in the reducer.

Reply:
I made the changes